selectrum-prescient.el

```diff
@@ -117,6 +117,73 @@ For use on `selectrum-candidate-selected-hook'."
     (remove-hook 'selectrum-candidate-inserted-hook
                  #'selectrum-prescient--remember)))
 
+;;;; Commands
+(defvar selectrum-prescient-filter-toggle-map (make-sparse-keymap)
+  "A keymap of commands for toggling Prescient filters in Selectrum.
+The toggling of commands is temporary and does not affect the
+default filtering settings determined by `prescient-filter-method'.")
+;; Create a binding similar to the Isearch toggles.
+(defvar selectrum-minibuffer-map)
+(define-key selectrum-minibuffer-map
+  "\M-s" selectrum-prescient-filter-toggle-map)
+
+(declare-function selectrum-exhibit "ext:selectrum")
+(defmacro selectrum--prescient-create-and-bind-toggle-command
+    (filter-type key-string)
+  "Create a command to toggle the use of FILTER-TYPE in Selectrum.
+FILTER-TYPE is an unquoted symbol which can be included in
+`prescient-filter-method'.  KEY-STRING is a string that can be
+passed to `kbd' which will be bound in
+`selectrum-prescient-filter-toggle-map'."
+  (let* ((filter-type-name (symbol-name filter-type))
+         (command-name (intern (concat "selectrum-prescient-toggle-"
+                                       filter-type-name))))
+    `(progn
+       (defun ,command-name
+           (arg) ; Arg list
+         ,(format
+           "Toggle the \"%s\" filter. With ARG, use only this filter."
+           filter-type-name)
+         (interactive "P")
+
+         (if arg
+             (setq-local prescient-filter-method
+                         (list (quote ,filter-type)))
+
+           ;; If needed, turn `prescient-filter-method' into a list of
+           ;; symbols.
+           (when (nlistp prescient-filter-method)
+             (setq-local prescient-filter-method
+                         (list prescient-filter-method)))
+
+           ;; Add or remove the filtering method from
+           ;; `prescient-filter-method' and tell the user what
+           ;; happened.
+           (if (memq (quote ,filter-type)
+                     prescient-filter-method)
+               (progn
+                 (setq-local prescient-filter-method
+                             (remove (quote ,filter-type)
+                                     prescient-filter-method))
+                 (message "%s filter toggled off."
+                          ,(capitalize filter-type-name)))
+             (setq-local prescient-filter-method
+                         (cons (quote ,filter-type)
+                               prescient-filter-method))
+             (message "%s filter toggled on."
+                      ,(capitalize filter-type-name))))
```

Comment:
Would it be a good idea to report the new value of `prescient-filter-method` after it's updated? It might be hard to keep track of which methods have been toggled on and off if there are multiple.